### ISSUE TITLE
unbound type symbol 'Item0.item'

### ISSUE THREAD

**Issue by yannickmoy** [xldenis/creusot#306]
I'm getting that error on the reproducer below:

```rust
extern crate creusot_contracts;
use creusot_contracts::*;

fn search() -> usize {
   #[invariant(dummy,true)]
   for i in 0..1 {
      if true {
         return i
      }
   }
   return 0
}
```

Do you have a suggestion to work around that issue? thanks

**Comment by sarsko** [comment]
I don’t believe iterators nor ranges are supported. I would attempt swapping to a while-loop.

**Comment by xldenis** [comment]
Yes @yannickmoy, iterators are not currently supported, though that will be changing very soon!

**Comment by xldenis** [comment]
The actual error message you are encountering is: https://github.com/xldenis/creusot/issues/213

**Comment by yannickmoy** [comment]
thanks both! making progress...

**Comment by xldenis** [comment]
You may find this example illuminating, I think you're familiar with the underlying function: https://github.com/xldenis/creusot/blob/master/creusot/tests/should_succeed/vector/01.rs

**Comment by yannickmoy** [comment]
I'm now using Why3 IDE to dig into unproved goals, and I have issues with CVC4 1.8 which systematically errs on `Unexpected token: 'char'`, is it because I'm not using the `stackify` branch of Why3? Things seem to work otherwise well with the default version installed with opam.

**Comment by xldenis** [comment]
Potentially, though now you simply need to install `why3` with `ocamlgraph` to have stackify. Could you upload a minimal example? I haven't seen this specific issue before.

**Comment by yannickmoy** [comment]
thanks, I installed `ocamlgraph` which recompiled `why3`, but I'm still getting that error from CVC4 1.8 on any file, say `vector/01.rs`:

```
(error "Parse Error: /tmp/why_91ca3f_01-T-all_zeroqtvc.smt2:1200.18: Unexpected token: 'char'.

  (declare-sort char 0)
                ^
")
smt::SmtEngine::resourceUnitsUsed, 275
```

Is there a way I can investigate?

**Comment by xldenis** [comment]
Ah, did you make sure to pin `why3` to the master branch like in the README? I believe I saw this issue before when Why3 didn't support CVC 1.8 properly.

**Comment by yannickmoy** [comment]
well spotted! it seemed to work without it, but that caused this issue with CVC4 indeed.

**Comment by xldenis** [comment]
I fixed this specific error message in #313, at least now you get an error in creusot explaining the proximate cause of the problem.
